stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1324,8 +1324,8 @@ julia> promote_leaf_eltypes(a)
 Complex{Float64}
 ```
 """
-promote_leaf_eltypes(x::Union{AbstractArray{T},Tuple{Vararg{T}}}) where {T<:Number} = T
-promote_leaf_eltypes(x::Union{AbstractArray{T},Tuple{Vararg{T}}}) where {T<:NumberArray} = eltype(T)
+promote_leaf_eltypes(x::Union{AbstractArray{T},Tuple{T,Vararg{T}}}) where {T<:Number} = T
+promote_leaf_eltypes(x::Union{AbstractArray{T},Tuple{T,Vararg{T}}}) where {T<:NumberArray} = eltype(T)
```

Comment:
[Copying it from my commit message] This change is only for making `Test.detect_ambiguities` and shouldn't introduce any change in working code (unless it depends on `MethodError`).


Before this change, we have:

```julia
julia> LinearAlgebra.promote_leaf_eltypes(())
ERROR: MethodError: LinearAlgebra.promote_leaf_eltypes(::Tuple{}) is ambiguous. Candidates:
...
```

With this change, we have:

```julia
julia> LinearAlgebra.promote_leaf_eltypes(())
Bool
```

Reply:
Actually, considering that we have `mapreduce(promote_leaf_eltypes, promote_type, x; init=Bool)` below, I suppose it was intended to return `Bool`?